File: compiler/testData/asJava/lightClasses/lightClassByPsi/k2/collections/mutable/mutableMapEntryInterface.java
public abstract class CMutableMapEntry /* test.CMutableMapEntry*/<KElem, VElem>  implements test.IMutableMapEntry<KElem, VElem> {
  public  CMutableMapEntry();//  .ctor()
}

public abstract class CMutableMapEntry2 /* test.CMutableMapEntry2*/<KElem, VElem>  implements test.IMutableMapEntry<KElem, VElem> {
  @java.lang.Override()
  @kotlin.IgnorableReturnValue()
  public VElem setValue(VElem);//  setValue(VElem)

  @java.lang.Override()
  public KElem getKey();//  getKey()

  @java.lang.Override()
  public VElem getValue();//  getValue()

  public  CMutableMapEntry2(@org.jetbrains.annotations.NotNull() @org.jetbrains.annotations.NotNull() test.IMutableMapEntry<KElem, VElem>);//  .ctor(@org.jetbrains.annotations.NotNull() test.IMutableMapEntry<KElem, VElem>)
}

public class CMutableMapEntry3 /* test.CMutableMapEntry3*/<KElem, VElem>  implements test.IMutableMapEntry<KElem, VElem> {
  @java.lang.Override()
  public KElem getKey();//  getKey()

  @java.lang.Override()
  public VElem getValue();//  getValue()

  @java.lang.Override()
  public VElem setValue(VElem);//  setValue(VElem)

  public  CMutableMapEntry3();//  .ctor()
}

public abstract interface IMutableMapEntry /* test.IMutableMapEntry*/<KElem, VElem>  extends java.util.Map.Entry<KElem, VElem>, kotlin.jvm.internal.markers.KMutableMap$Entry {
}

File: compiler/testData/asJava/lightClasses/lightClassByPsi/collections/mutable/mutableMapEntryInterface.kt
// WITH_STDLIB
package test

interface IMutableMapEntry<KElem, VElem> : MutableMap.MutableEntry<KElem, VElem>

abstract class CMutableMapEntry<KElem, VElem> : IMutableMapEntry<KElem, VElem>

abstract class CMutableMapEntry2<KElem, VElem>(d: IMutableMapEntry<KElem, VElem>) : IMutableMapEntry<KElem, VElem> by d

open class CMutableMapEntry3<KElem, VElem> : IMutableMapEntry<KElem, VElem> {
    override fun setValue(newValue: VElem): VElem {
        TODO("Not yet implemented")
    }

    override val key: KElem
        get() = TODO("Not yet implemented")
    override val value: VElem
        get() = TODO("Not yet implemented")
}